src/components/profile-rewards/profile-rewards.component.tsx

```diff
@@ -1,20 +1,50 @@
-import React from 'react';
+import React, { useState }  from 'react';
 import './profile-rewards.styles.scss';
 
-import productImg from '../../assets/images/img-product.png';
+import { PeiliUserConsumer } from '../../context/userContext';
 
-const Rewards = () => {
+const Rewards = (props: any) => {
+  const [ isRedeemed, redeem ] = useState(false);
+  const [ errorRedeem, setError ] = useState(false);
   return (
-    <div className="rewards">
-      <div className="rewards-img-container">
-        <img src={productImg} alt="product" />
-      </div>
-      <div className="rewards-info">
-        <p className="rewards-subtitle">ELIXIA exclusive 3-month membership</p>
-        <p className="rewards-exp">100 EXP</p>
-        <button className="rewards-button">redeem</button>
-      </div>
-    </div>
+    <PeiliUserConsumer>
+      {(value: any) => {
+        const { user, substractEXP } = value;
+        const { reward } = props;
+        const redeemReward = (exp: any) => {
+          if (user.exp >= exp) {
+            redeem(true); // change button style
+            substractEXP(exp); // update user data
+          } else {
+            setError(true); // render error in reward card
+          }
+        };
+        return (
+          <div className="rewards">
+            <div className="rewards-img-container">
+              <img src={reward.image} alt="product" />
+            </div>
+            <div className="rewards-info">
+              <p className="rewards-subtitle">{reward.description}</p>
+              <p className="rewards-exp">{reward.exp} EXP</p>
+              <p 
+                style={{display: errorRedeem ? 'block' : 'none', color: 'red', marginBottom: '10px'}}
+              >
+                You don't have enough EXP to redeem
+              </p>
+              <button 
+                onClick={() => redeemReward(reward.exp)}
+                className="rewards-button" 
+                style={{backgroundColor: isRedeemed ? '#000' : '#fff', color: isRedeemed ? '#fff' : '#000'}}
```

Comment:
Every component has its own scss file to define its styles. Let's move all the inline styles there.

Reply:
resolved